analyzers/tests/SonarAnalyzer.UnitTest/Extensions/AssignmentExpressionSyntaxExtensionsTests.cs

```diff
@@ -0,0 +1,364 @@
+            var code = "(int, int) tuple; tuple =  (1, 2);";
+            var mapping = ParseAssignmentExpression(code).MapAssignmentArguments();
+            mapping.Should().BeEquivalentTo(new[]
+                {
+                    new
+                    {
+                        Left = new { Identifier = new { Text = "tuple" } },
+                        Right = new
+                        {
+                            Arguments = new[]
+                            {
+                                new { Expression = new { Token = new { Text = "1" } } },
+                                new { Expression = new { Token = new { Text = "2" } } },
+                            }
+                        },
+                    },
+                });
+        }
+
+        [TestMethod]
+        public void AssignmentExpressionSyntaxExtensions_SimpleDeconstructionAssignment()
+        {
+            var code = "var (x, y) =  (1, 2);";
+            var mapping = ParseAssignmentExpression(code).MapAssignmentArguments();
+            mapping.Should().BeEquivalentTo(new[]
+                {
+                    new
+                    {
+                        Left = new { Identifier = new { Text = "x" } },
+                        Right = new { Token = new { Text = "1" } },
+                    },
+                    new
+                    {
+                        Left = new { Identifier = new { Text = "y" } },
+                        Right = new { Token = new { Text = "2" } },
+                    },
+                });
+        }
+
+        [TestMethod]
+        public void AssignmentExpressionSyntaxExtensions_NestedDeconstructionAssignment()
+        {
+            var code = "var (a, (b, c), d) =  (1, (2, 3), 4);";
+            var mapping = ParseAssignmentExpression(code).MapAssignmentArguments();
+            mapping.Should().BeEquivalentTo(new[]
+                {
+                    new
+                    {
+                        Left = new { Identifier = new { Text = "a" } },
+                        Right = new { Token = new { Text = "1" } },
+                    },
+                    new
+                    {
+                        Left = new { Identifier = new { Text = "b" } },
+                        Right = new { Token = new { Text = "2" } },
+                    },
+                    new
+                    {
+                        Left = new { Identifier = new { Text = "c" } },
+                        Right = new { Token = new { Text = "3" } },
+                    },
+                    new
+                    {
+                        Left = new { Identifier = new { Text = "d" } },
+                        Right = new { Token = new { Text = "4" } },
+                    },
+                });
+        }
+
+        [TestMethod]
+        public void AssignmentExpressionSyntaxExtensions_MisalignedDeconstructionAssignment()
+        {
+            var code = "var (a, (b, c, d)) =  (1, (2, 3), 4);";
```

Comment:
Can't we somehow detect that they are not matching and return the initial state or an empty Array? I fear that we will start raising issues for non compiling code inside the IDE in these cases.